src/QMCWaveFunctions/lcao/LCAOrbitalSetWithCorrection.h

```diff
@@ -0,0 +1,78 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Jeongnim Kim, jeongnim.kim@intel.com, Intel Corp.
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_SOA_LINEARCOMIBINATIONORBITALSET_WITH_CORRECTION_TEMP_H
+#define QMCPLUSPLUS_SOA_LINEARCOMIBINATIONORBITALSET_WITH_CORRECTION_TEMP_H
+
+#include "QMCWaveFunctions/SPOSet.h"
+#include "QMCWaveFunctions/BasisSetBase.h"
+#include "QMCWaveFunctions/lcao/LCAOrbitalSet.h"
+#include "QMCWaveFunctions/lcao/SoaCuspCorrectionBasisSet.h"
+
+
+namespace qmcplusplus
+{
+/** class to add cusp correction to LCAOrbitalSet.
+   *
+   */
+struct LCAOrbitalSetWithCorrection : public LCAOrbitalSet
+{
+  SoaCuspCorrection* cusp;
```

Comment:
I still don't like using pointer. Is it possible to have it non-pointer? Does it have pointers internally? Can we rely on copy constructor?

Reply:
I'll have to look through the makeClone methods to see if this is possible.

Reply:
I mean only SoaCuspCorrection not the upper levels.

Reply:
std::unique_ptr or other smart pointer? It could make the SoaCuspCorrection lifetime more explicit.
 
Since this is a new class can we explicitly specify the access?
```C++
class LCAOribtalSetWithCorrections : public LCAOOrbitalSet
{
public:
...
```

Reply:
I think it is possible to refer to SoaCuspCorrection by value, and depend on the copy constructor to carry out the makeClone operations.